airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/token.py

```diff
@@ -115,3 +118,121 @@ def token(self) -> str:
+        session_token_response_key (Union[InterpolatedString, str]): Key for retrieving session token from api response
+    """
+
+    api_url: Union[InterpolatedString, str]
+    header: Union[InterpolatedString, str]
+    session_token: Union[InterpolatedString, str]
+    session_token_response_key: Union[InterpolatedString, str]
+    username: Union[InterpolatedString, str]
+    config: Config
+    options: InitVar[Mapping[str, Any]]
+    password: Union[InterpolatedString, str] = ""
+
+    def __post_init__(self, options):
+        self._username = InterpolatedString.create(self.username, options=options)
+        self._password = InterpolatedString.create(self.password, options=options)
+        self._api_url = InterpolatedString.create(self.api_url, options=options)
+        self._header = InterpolatedString.create(self.header, options=options)
+        self._session_token = InterpolatedString.create(self.session_token, options=options)
+        self._session_token_response_key = InterpolatedString.create(self.session_token_response_key, options=options)
+
+        self.logger = logging.getLogger("airbyte")
+
+    @property
+    def auth_header(self) -> str:
+        return self._header.eval(self.config)
+
+    @property
+    def token(self) -> str:
+        if self._session_token.eval(self.config):
+            if self.is_valid_session_token():
+                return self._session_token.eval(self.config)
+        if self._password.eval(self.config) and self._username.eval(self.config):
+            username = self._username.eval(self.config)
+            password = self._password.eval(self.config)
+            session_token_response_key = self._session_token_response_key.eval(self.config)
+            api_url = f"{self._api_url.eval(self.config)}session"
+
+            self.logger.info("Using generated session token by username and password")
+            return get_new_session_token(api_url, username, password, session_token_response_key)
+
+        raise ConnectionError("Invalid credentials: session token is not valid or provide username and password")
+
+    def is_valid_session_token(self) -> bool:
+        try:
+            response = requests.get(
+                f"{self._api_url.eval(self.config)}user/current", headers={self.auth_header: self._session_token.eval(self.config)}
+            )
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == requests.codes["unauthorized"]:
+                self.logger.info(f"Unable to connect by session token from config due to {str(e)}")
+                return False
+            else:
+                raise ConnectionError(f"Error while validating session token: {e}")
+        if response.ok:
+            json_response = response.json()
+            self.logger.info(
+                f"Connection check for source successful for {json_response['common_name']} login at {json_response['last_login']}"
```

Comment:
These two fields need to be configurable because the endpoints won't always return responses with "common_name" and "last_login" fields

Reply:
alternatively, it would be fine not to log this much details here

Reply:
done